service/storage_service.cc

```diff
@@ -1784,6 +1786,47 @@ class topology_coordinator {
         }
 
         switch (*tstate) {
+            case topology::transition_state::join_group0: {
+                auto node = get_node_to_work_on(std::move(guard));
+                switch (node.rs->state) {
+                    case node_state::bootstrapping: {
+                        assert(node.rs->ring);
+                        auto tmptr = get_token_metadata_ptr();
+                        auto [gen_uuid, guard, mutation] = co_await prepare_and_broadcast_cdc_generation_data(
```

Comment:
Can you explain why you decided to move CDC here, but not token assignment? I think both can stay where they are now.

Reply:
We have an existing assertion in `load_topology_state` which checks that transitioning nodes (except for `left_token_ring` state) have tokens assigned, so I kept the logic that assigns the token in the first step.

As for the CDC part, I didn't want `prepare_and_broadcast_cdc_generation_data` to be called while a handshake is in progress. I wanted the handshake to take as little time as possible to reduce the chance of failure, and `prepare_and_broadcast_cdc_generation_data` may potentially transfer a large chunk of data.

Reply:
I see.